bridges/snowbridge/primitives/router/src/inbound/envelope.rs

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+use snowbridge_core::{inbound::Log, ChannelId};
+
+use sp_core::{RuntimeDebug, H160, H256};
+use sp_std::prelude::*;
+
+use alloy_primitives::B256;
+use alloy_sol_types::{sol, SolEvent};
+
+sol! {
```

Comment:
what is this macro?

Reply:
`sol!`  macro takes an event declaration in solidity and provides helper functions and also struct to work with that event data. Like encoding/decoding etc.